cli/command/cli.go

```diff
@@ -45,7 +45,7 @@ type DockerCli struct {
 func (cli *DockerCli) HasExperimental() bool {
 	if cli.hasExperimental == nil {
 		if cli.client == nil {
-			cli.Initialize(cliflags.NewClientOptions())
+			return false
 		}
 		enabled := false
 		cli.hasExperimental = &enabled
```

Comment:
This is now called from only one place, after initializing the client.  Maybe we could remove `HasExperimental()` and instead just call `Ping()` directly?

I think it would still be only a single function call. The rest of this code is just for managing state, and we don't need the state if we're only looking at it once.